install/threesdk/cli.py

```diff
@@ -110,15 +110,76 @@ def version():
     Print version number
     """
     print(f"3sdk {_sdk.__version__}")
-    # from InstallTools import get_newest_version
-    dic = _sdk.InstallTools.Tools.get_newest_version()
     # campare with __version__
-    if not dic["up_to_date"]:
+    if not _sdk.InstallTools.Tools.is_latest_release():
+        dic = _sdk.InstallTools.Tools.get_latest_release()
         print(
-            f"Your SDK version is not up-to-date. Newest release is {dic['last_release']}\nPlease visit: {dic['last_release_url']}"
+            f"Your SDK version is not up-to-date. Newest release is {dic['latest_release']}\nPlease visit: {dic['latest_release_url']}\nOr run '3sdk --update'"
         )
 
 
+def update():
+    """
+    """
+    from threesdk import container
+    from threesdk.InstallTools import JumpscaleInstaller, MyEnv, Tools, DockerFactory
+
+    if not Tools.is_latest_release():
+        installer = JumpscaleInstaller()
+        containers_names = DockerFactory.containers()
+        host_mount = False
+        # pull all repos in containers
+        for item in containers_names:
+            name = item.name
+            print("Updating repositories in container ", name, "...")
+            container._containers.assert_container(name)
+            c = DockerFactory.container_get(name=name)
+            if not c.mount_code_exists:
+                container_executor = c.executor
+                installer.repos_get(pull=True, executor=container_executor)
+            else:
+                host_mount = True
+        if host_mount:
+            installer.repos_get(pull=True)
+
+        import requests
+
+        # Update binary for host
+        print("Downloading 3sdk binary...")
+        latest_release = Tools.get_latest_release()
+        with requests.get(latest_release["download_link"], allow_redirects=True, stream=True) as r:
+            if MyEnv.platform_is_linux or MyEnv.platform_is_osx:
+                local_filename = "/tmp/3sdk"
+                with open(local_filename, "wb") as f:
+                    shutil.copyfileobj(r.raw, f)
+                f.close()
```

Comment:
the file does not need to be closed the with clause does this.
the remote connection however isn't so this should be r.closer()

Reply:
Closed the remote connection, thanks Jo